Code/Framework/AzFramework/AzFramework/Input/Channels/InputChannelId.h

```diff
@@ -39,21 +40,28 @@ namespace AzFramework
 
         ////////////////////////////////////////////////////////////////////////////////////////////
         //! Constructor
-        //! \param[in] name Name of the input channel (will be truncated if exceeds MAX_NAME_LENGTH)
-        explicit InputChannelId(const char* name = "");
-
-        ////////////////////////////////////////////////////////////////////////////////////////////
-        //! Copy constructor
-        //! \param[in] other Another instance of the class to copy from
-        InputChannelId(const InputChannelId& other);
-
-        ////////////////////////////////////////////////////////////////////////////////////////////
-        //! Copy assignment operator
-        //! \param[in] other Another instance of the class to copy from
-        InputChannelId& operator=(const InputChannelId& other);
+        //! \param[in] name Name of the input channel (will be ignored if exceeds MAX_NAME_LENGTH)
+        explicit constexpr InputChannelId(AZStd::string_view name = "")
+            : m_name(name)
+            , m_crc32(name)
+        {
+        }
 
-        ////////////////////////////////////////////////////////////////////////////////////////////
-        //! Default destructor
+        constexpr InputChannelId(const InputChannelId& other) = default;
+        constexpr InputChannelId(InputChannelId&& other) = default;
+        constexpr InputChannelId& operator=(const InputChannelId& other)
+        {
+            m_name = other.m_name;
+            m_crc32 = other.m_crc32;
+            return *this;
+        }
+        constexpr InputChannelId& operator=(InputChannelId&& other)
+        {
+            m_name = AZStd::move(other.m_name);
+            m_crc32 = AZStd::move(other.m_crc32);
+            other.m_crc32 = 0;
+            return *this;
+        }
```

Comment:
Just a thought, but could we just 'default' these instead now?

Reply:
Or better yet, just replace these and the two lines above with:

```
        ////////////////////////////////////////////////////////////////////////////////////////////
        // Default copying and moving
        AZ_DEFAULT_COPY_MOVE(InputChannelId);
```

Reply:
I thought the Ubuntu nodes were updated to 20.04 LTS and were using clang-12?

Reply:
I believe that they have both 12 and 6 installed now, but I would check with Infra first. If we decide to make 12 the minimum version of clang, then I can update that in the Linux setup guide as well

Reply:
I think we should make the minimum clang12.